device/transport/amqp/test/_amqp_twin_client_test.js

```diff
@@ -93,687 +113,448 @@ describe('AmqpTwinClient', function () {
-      var dummyPostHandler = function() {};
-      var newTwinReceiver = new AmqpTwinClient(fakeAuthenticationProvider, amqpClient);
-      var sendCalled = sinon.spy(amqpClient.fakeSenderLink, 'send');
-      newTwinReceiver.on('subscribed', function(event) {
-        if (event.eventName === 'post') {
-          var deleteResponse = new Message();
-          newTwinReceiver.on('response', function() {});
-          newTwinReceiver.removeListener('post', dummyPostHandler);
-          assert(sendCalled.calledTwice, 'sendTwinRequest invoked incorrect number of times');
-          assert.equal(amqpClient.lastSendMessage.messageAnnotations['operation'], 'DELETE', 'inappropriate operation sent');
-          assert.equal(amqpClient.lastSendMessage.messageAnnotations['resource'], resource, 'inappropriate resource sent');
-          deleteResponse.correlationId = amqpClient.lastSendMessage.properties.correlationId;
-          deleteResponse.data = '';
-          amqpClient.fakeReceiverLink.emit('message', deleteResponse);
-          responseDone();
-        }
-      });
-      newTwinReceiver.on('post', dummyPostHandler);
-      var postResponse = new Message();
-      postResponse.correlationId = amqpClient.lastSendMessage.properties.correlationId;
-      postResponse.data = '';
-      amqpClient.fakeReceiverLink.emit('message', postResponse);
+    it('sends a message with no resource annotation', function () {
+      twinClient.getTwin(function () {});
+      assert.isUndefined(fakeSenderLink.send.firstCall.args[0].messageAnnotations.resource);
     });
 
-  });
+    it('sends a message with a correlation id that is an integer', function () {
+      twinClient.getTwin(function () {});
+      var correlationId = parseInt(fakeSenderLink.send.firstCall.args[0].properties.correlationId);
+      assert.isNumber(correlationId);
+      assert.isNotNaN(correlationId);
+    });
 
-  describe('#errors', function() {
-    /* Tests_SRS_NODE_DEVICE_AMQP_TWIN_06_022: [If an error occurs on establishing the upstream or downstream link then the `error` event shall be emitted.] */
-    [
-      { direction: 'sender', provider: AmqpProviderAttachSenderFails },
-      { direction: 'receiver', provider: AmqpProviderAttachReceiverFails }
-    ].forEach(function(operation) {
-      it('twin receiver emits an error if attaching ' + operation.direction + ' link fails', function(testDone) {
-        var amqpClient = new operation.provider();
-        var newTwinReceiver = new AmqpTwinClient(fakeAuthenticationProvider, amqpClient);
-        newTwinReceiver.on('error', function(err) {
-          assert.equal(err.constructor.name, 'UnauthorizedError');
-          testDone();
-        });
-        newTwinReceiver.on('response', function () {});
-      });
+    it('sends a message with an empty body', function () {
+      twinClient.getTwin(function () {});
+      assert.strictEqual(fakeSenderLink.send.firstCall.args[0].body, ' ');
     });
```

Comment:
Do you get paid by the test?  These 4 could be 1.

Reply:
add nit:

---
In reply to: [171690404](https://github.com/pierreca/azure-iot-sdk-node/pull/1#discussion_r171690404) [](ancestors = 171690404)

Reply:
i always wonder about this: we're testing 4 different things technically. one could break independently of the others. but at the same time having 4 tests instead of 1 doesn't bring a lot of value. I guess I'll just make this one bigger test

---
In reply to: [171690468](https://github.com/pierreca/azure-iot-sdk-node/pull/1#discussion_r171690468) [](ancestors = 171690468,171690404)